src/apic-extension/azext_apic_extension/tests/latest/utils.py

```diff
@@ -380,3 +381,64 @@ def _get_apic_environment(self, **kwargs):
             template = 'To create an API Center Deployment an API Center Environment is required. Please add ' \
                        'decorator @{} in front of this preparer.'
             raise CliTestError(template.format(ApicEnvironmentPreparer.__name__))
+        
+class ApimServicePreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
+    def __init__(self, name_prefix='clitest', length=24,
+                 parameter_name='apim_name', resource_group_parameter_name='resource_group', key='apim_name',
+                 enable_system_assigned_identity=False):
+        super(ApimServicePreparer, self).__init__(name_prefix, length)
+        self.cli_ctx = get_dummy_cli()
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.parameter_name = parameter_name
+        self.enable_system_assigned_identity = enable_system_assigned_identity
+        self.key = key
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+
+        # Only setup APIM in live mode
+        if self.test_class_instance.is_live:
+            # Get system assigned identity id for API Center
+            apic_service = self.test_class_instance.cmd('az apic show -g {} -n {}'.format(group, self.test_class_instance.kwargs['s'])).get_output_in_json()
```

Comment:
The `ApicServicePreparer` may set the service name to different parameter. You can refer other preparers to define a default parameter name and let people customize the name when needed.